### PR TITLE
Fix unstable / randomly failing IT methods in AuthenticationRestControllerIT

### PR DESCRIPTION
**Pull request by tdonohue**
## Description
As other developers will have noticed, we have had recent "random" test failures from two `AuthenticationRestControllerIT` methods: `testStatusShibAuthenticatedWithCookie()` and `testStatusPasswordAuthenticatedWithCookie()`.

The random failures look like this:
```
Error:  Failures: 
Error:    AuthenticationRestControllerIT.testStatusPasswordAuthenticatedWithCookie:309 Response header 'Authorization' expected:<*** but was:<***
```

This PR fixes those random failures by correcting the tests to provide a more accurate & more stable behavior.

The random failures were caused by invalid tests which were not anticipating that JWT auth tokens _may actually change_ when they are refreshed or reloaded.   Our JWTs all have an expiration date assigned, and that expiration date is based on the token creation or refresh date.  Therefore, if even *one second* has passed, the expiration date will be one second later, and the JWT auth tokens will be different.  This behavior is **expected behavior** and we even have an existing test to prove it, see `testTwoAuthenticationTokens()`

To fix the random failures, I updated several tests to no longer perform an exact string comparison between two JWTs.  Instead, a new `tokenClaimsEqual()` method was added to compare the JWT claims to ensure they are identical (while ignoring changes to the expiration date) & updating affected tests to use that method.  I also updated the existing `testTwoAuthenticationTokens()` method to similarly use that method.

Finally, I made some minor updates to other IT methods as I noticed others were not properly parsing out the JWT tokens from headers (this was discovered when I installed this PR alongside #8030, as it was obvious from the logs that other methods were not behaving properly).

## Instructions for Reviewers
Review the code changes.  Verify the ITs pass (they should..I've run these tests 10 times in a row locally, and have not seen any more of the random IT failures).

### CODE REVIEW

**Review by LucaGiamminonni — APPROVED** [review]
Hi @tdonohue, I have reviewed the code and the changes seem ok to me. I have tried to run the tests locally multiple times and have not had any problems. So for me this PR can be merged.

**Comment by tdonohue** [comment]
Thanks @LucaGiamminonni !

I had to just resolve a very minor merge conflict in this PR after merging #8032.  It doesn't change the code in this PR at all though, so I'll just wait for our GitHub CI process to succeed one last time before merging this.

**Comment by tdonohue** [comment]
Tests succeeded after rebase.  Merging with +1 approval.

**Pull request merged**
